### ISSUE TITLE
new image recover problem

### ISSUE THREAD

**Issue by Xiyor** [rohitrango/automatic-watermark-detection#3]
Hi, I had download 400 fotolia images, run main.py script, and pickled dump W and alpha. However, how to use W and alpha to recover a watermarked image? Directly use I(p) = (J(p) − α(p)W(p))/(1 − α(p))? I tried, but the performance is bad. Could you give me some suggestion?
.

**Comment by rohitrango** [comment]
As they mention it in their paper, you have to iteratively reconstruct I given all the other variables (including the auxiliary ones). Simply plugging in the formula won't really work.
I hope this clarifies your question.

**Comment by Xiyor** [comment]
@rohitrango 
As section3.3 in paper mention: " Once we have the solution for W and α, we can use them
to remove the watermark from any new image marked by it, without the need to run the entire pipeline again. "  and this sentence seems to tell us, get W and alpha, we can directly recover image. 
According to you, How we can recover a image by got W and alpha?

**Comment by summer1914** [comment]
I have the same problem.
Rohitrango could you please show us how to use the W and alpha to recover a image which is watermarked by the same one we trained ?

**Comment by rohitrango** [comment]
Please refer to the supplementary material of the paper.

**Comment by Microstrong0305** [comment]
@rohitrango 
Hi, I read the paper and supplementary material carefully and found three places where performance can be improved. (1) Images used to estimate the initial matted watermark (Sec. 3.2). My understanding is that the number of pictures equals the size of dataset. I had dowmload 630 fotolia images. (2) images used in the multi-images matting step (Sec. 3.2). Corresponding to your code is "Jt = J[:25]"，You select 25 pictures to optimize W and α. (3) The number of iterations is set when the algorithm is optimized. Corresponding to your code is "def solve_images(J, W_m, alpha, W_init, gamma=1, beta=1, lambda_w=0.005, lambda_i=1, lambda_a=0.01, iters=4)". For the above three points, I have adjusted parameters. However, I found that the performance was not very good. Could you give me some suggestion?
My results are as follows:
![fotolia_75353029](https://user-images.githubusercontent.com/20468401/38180487-8c91710a-365f-11e8-8ba6-94674722d63c.jpg)

![fotolia_137840668](https://user-images.githubusercontent.com/20468401/38180478-7da8c24c-365f-11e8-9b5a-5d4dacfb5f9b.jpg)

My code are as follows:
# put W and α into an array of (500,500,3)
W_new = np.zeros((500,500,3))
alpha1_new = np.zeros((500,500,3)
W_new[218:281,189:309,:] = W
alpha1_new[218:281,189:309,:] = alpha1
# Remove watermark
Output = (img - alpha1_new * W_new) / (np.ones((500, 500, 3)) - alpha1_new)

**Comment by rohitrango** [comment]
Yeah, so the supplementary actually doesn't use the last formula that you mentioned. Instead, it uses the reconstruction using IRLS.
